### PR TITLE
feat: introduce dev-maker

### LINKED ISSUE #2260: Implement dev-maker
We should implement a "dev" maker which is in the main repository.

Ideally, the dev-maker does not require an internet connection. We could either just do static prices or put a price.json file which contains historic data. This way we get some "randomness". 

Having this dev-maker would simplify breaking api changes as one does not have to work on two repos at the same time which depend on each other.

### PR DESCRIPTION
**Pull request by bonomat**
resolves #2260

### CODE REVIEW

File: crates/dev-maker/fetch_rates.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+API_URL="https://www.bitmex.com/api/v1"
+
+SYMBOL="XBTUSD"
+
+# Set start and end timestamps for the date range (March 20, 2023, to March 20, 2024)
+START_TIMESTAMP="2023-03-20T00:00:00.000Z"
+END_TIMESTAMP="2024-03-20T00:00:00.000Z"
+
+# Construct the request URL for hourly data
+REQUEST_URL="${API_URL}/trade/bucketed?symbol=${SYMBOL}&binSize=1h&partial=false&count=1000&reverse=false&startTime=${START_TIMESTAMP}&endTime=${END_TIMESTAMP}"
```
Comment (holzeis) [inline crates/dev-maker/fetch_rates.sh:14]: nit: We only fetch the first 1000 prices here, which gets us to April 2023. We might want to have more recent prices?

File: crates/dev-maker/fetch_rates.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+# A script to fetch historic rates from Bitmex between a specifc time window
+
+# BitMEX API base URL
+API_URL="https://www.bitmex.com/api/v1"
+
+SYMBOL="XBTUSD"
+
+# Set start and end timestamps for the date range (March 20, 2023, to March 20, 2024)
+START_TIMESTAMP="2023-03-20T00:00:00.000Z"
```
Comment (holzeis) [inline crates/dev-maker/fetch_rates.sh:10]: nit: How about making the script more generic by automatically calculating the start and the end timestamp based on the current date?

File: justfile [hunk 3]
```diff
@@ -302,9 +302,7 @@ run_maker_args := if os() == "linux" {
 }
 
 maker args="":
-    # we always delete the old container first as otherwise we might get an error if the container still exists
-    docker rm -f maker || true
-    docker run {{run_maker_args}}
+    cargo run --bin dev-maker
```
Comment (holzeis) [inline justfile:305]: 🤔 I guess we will have to add the dev makers pubkey to the `whitelisted_makers` in the `test-coordinator-settings.toml`

**Review by holzeis — COMMENTED** [review]
Nice 👍 Thank you!

File: crates/dev-maker/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    historic_rates.sort_by(|a, b| a.timestamp.cmp(&b.timestamp));
+
+    loop {
+        for historic_rate in &historic_rates {
+            post_order(
+                client.clone(),
+                secret_key,
+                public_key,
+                Direction::Short,
+                historic_rate.open + Decimal::from(1),
```
[the remark this answers is not in the document]
Comment (bonomat) [reply crates/dev-maker/src/main.rs:45]: No, I think this is correct... the order is my indication to go _short_, so you would go _long_, or differently said, I'm selling this to you for this price. Hence, I charge `1` more.

File: crates/dev-maker/fetch_rates.sh (hunk 1 above)
Comment (bonomat) [reply crates/dev-maker/fetch_rates.sh:14]: feel free to update it, I think for local testing it doesn't matter much.

File: justfile (hunk 3 above)
Comment (bonomat) [reply justfile:305]: For local dev setup it shouldn't be needed because whitelisting should be disabled.

**Comment by bonomat** [comment]
> Nice 👍 Thank you!

Feel free to approve things ;)

File: crates/dev-maker/fetch_rates.sh (hunk 2 above)
Comment (bonomat) [reply crates/dev-maker/fetch_rates.sh:10]: I asked chat-gtp and the `date` library on linux works differently than on mac, hence, I hardcoded it 😬

**Pull request merged**